src/writeData/components/AddPluginToConfiguration.tsx

```diff
@@ -30,84 +41,121 @@ interface AddPluginToConfigurationCTAProps {
   >
 }
 
-export const AddPluginToConfigurationCTA: FC<AddPluginToConfigurationCTAProps> = ({
-  contentID,
-  history,
-  orgID,
-  thumbnail,
-  pageContent,
-}) => (
-  <>
-    <div className="write-data--details">
-      <Grid>
-        <Grid.Row>
-          <Grid.Column
-            widthXS={Columns.Twelve}
-            widthMD={Columns.Six}
-            widthLG={Columns.Three}
-          >
-            <div className="write-data--details-thumbnail">{thumbnail}</div>
-            <Dropdown
-              button={(active, onClick) => (
-                <Dropdown.Button
-                  active={active}
-                  color={ComponentColor.Primary}
-                  style={{width: '100%'}}
-                  onClick={onClick}
-                >
-                  Use this plugin
-                </Dropdown.Button>
-              )}
-              className="use-plugin--telegraf-configuration"
-              menu={onCollapse => (
-                <Dropdown.Menu onCollapse={onCollapse}>
-                  <Dropdown.Item
-                    key="Create-new-configuration-telegraf-plugin"
-                    value="Create a new configuration"
-                    onClick={() => {
-                      history.push(
-                        `/${ORGS}/${orgID}/load-data/${TELEGRAF_PLUGINS}/${contentID}/new`
-                      )
-                    }}
-                    selected={false}
-                    testID="create-new-configuration-from-plugin--dropdown-item"
-                  >
-                    <span>Create a new configuration</span>
-                  </Dropdown.Item>
-                  <Dropdown.Item
-                    key="Add-to-existing-configuration-telegraf-plugin"
-                    value="Add to an existing configuration"
-                    onClick={() => {}}
-                    selected={false}
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & AddPluginToConfigurationCTAProps
+
+const AddPluginToConfigurationCTAComponent: FC<Props> = props => {
+  const {
+    contentID,
+    fetchTelegrafs,
+    history,
+    orgID,
+    pageContent,
+    telegrafs,
+    thumbnail,
+  } = props
+
+  useEffect(() => {
+    fetchTelegrafs()
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <>
+      <div className="write-data--details">
+        <Grid>
+          <Grid.Row>
+            <Grid.Column
+              widthXS={Columns.Twelve}
+              widthMD={Columns.Six}
+              widthLG={Columns.Three}
+            >
+              <div className="write-data--details-thumbnail">{thumbnail}</div>
+              <Dropdown
+                button={(active, onClick) => (
+                  <Dropdown.Button
+                    active={active}
+                    color={ComponentColor.Primary}
+                    style={{width: '100%'}}
```

Comment:
so i recently learned passing props as objects is a big cause of re-renders. since this is a completely static style, can you pull `{width: '100%'}` outside of this component and store it as a variable?

something like:

```ts
const fullWidth = {width: '100%'}
const AddPluginToConfigurationCTAComponent: FC<Props> = props => {
```

and then use it like:

```ts
                  <Dropdown.Button
                    active={active}
                    color={ComponentColor.Primary}
                    style={fullWidth}
```

that'll keep this from always re-rendering, since it won't need to [recreate the object on every render](https://www.debugbear.com/blog/react-rerenders#passing-objects-as-props)

Reply:
Updated.